print_format/request_for_quotation.jinja

```diff
@@ -8,12 +8,18 @@
     }
 </style>
 
-{% set company_address_name = frappe.db.get_value("Dynamic Link", filters={"link_doctype": "Company", "link_name": doc.company, "parenttype": "Address"}, fieldname="parent") %}
-{% set cmp_addr = frappe.get_doc("Address", company_address_name) %}
+{% set company_addresses = frappe.get_all("Dynamic Link", filters={"link_doctype": "Company", "link_name": doc.company, "parenttype": "Address"}, fields=["parent"]) %}
+{% set cmp_addr = frappe.get_doc("Address", company_addresses[0]['parent']) %}
 {% set company = frappe.get_doc("Company", doc.company) %}
 
-{% set supplier_address_name = frappe.db.get_value("Dynamic Link", filters={"link_doctype": "Supplier", "link_name": doc.supplier, "parenttype": "Address"}, fieldname="parent") %}
-{% set pay_addr = frappe.get_doc("Address", supplier_address_name) %}
+{% if doc.vendor %}
+    {% set supplier_handle = doc.vendor %}
+{% else %}
+    {% set supplier_handle = doc.suppliers[0].supplier %}
+{% endif %}
+{% set supplier_doc = frappe.get_doc("Supplier", supplier_handle) %}
+{% set supplier_addresses = frappe.get_all("Dynamic Link", filters={"link_doctype": "Supplier", "link_name": supplier_handle, "parenttype": "Address"}, fields=["parent"]) %}
+{% set pay_addr = frappe.get_doc("Address", supplier_addresses[0]['parent']) %}
```

Comment:
This part doesn't need to change (apart from replacing `doc.supplier` with `supplier_doc.name`).

Reply:
```suggestion
{% set supplier_doc = frappe.get_doc("Supplier", supplier_handle) %}
{% set supplier_address_name = frappe.db.get_value("Dynamic Link", filters={"link_doctype": "Supplier", "link_name": supplier_doc.name, "parenttype": "Address"}, fieldname="parent") %}
{% set pay_addr = frappe.get_doc("Address", supplier_address_name) %}
```
Thanks. I tested it and it works. 
I just needed to additionally set the variable supplier_doc